Frends.Community.ActiveMQ/ActiveMQTasks.cs

```diff
@@ -39,41 +44,54 @@ public static async Task<Result> Consume([PropertyTab] Input input, [PropertyTab
+                            if (rawMsg == null) break;
+
+                            if (rawMsg is ITextMessage textMessage)
                             {
                                 messages.Add(new Message("Text", textMessage.Text));
                                 readNextMessage = true;
                             }
-                            else if (task.Result is IStreamMessage streamMessage)
+                            else if (rawMsg is IStreamMessage)
                             {
                                 messages.Add(new Message("Stream", "Stream message are not supported"));
                                 readNextMessage = true;
                             }
-                            else if (task.Result is IBytesMessage bytesMessage)
+                            else if (rawMsg is IBytesMessage bytesMessage)
                             {
                                 messages.Add(new Message("Bytes", bytesMessage.Content));
                                 readNextMessage = true;
                             }
-                            else if (task.Result is IMapMessage mapMessage)
+                            else if (rawMsg is IMapMessage)
                             {
                                 messages.Add(new Message("Map", "Map messages are not supported"));
                                 readNextMessage = true;
                             }
-                            else if (task.Result is IObjectMessage objectMessage)
+                            else if (rawMsg is IObjectMessage)
                             {
                                 messages.Add(new Message("Object", "Object messages are not supported"));
                                 readNextMessage = true;
                             }
-                            else if (task.Result is null)
-                                break;
                             else
-                                messages.Add(new Message("Unknown message type", "Unknown message type: " + task.Result.GetType().Name));
+                            {
+                                messages.Add(new Message("Unknown message type", "Unknown message type: " + rawMsg.GetType().Name));
+                                readNextMessage = true;
+                            }
+
+                            if (options.Acknowledge == AcknowledgeBehavior.OnSuccess)
+                                receivedRawMessages.Add(rawMsg);
 
                             if (options.MaxMessagesToConsume > 0 &&
                                 messages.Count >= options.MaxMessagesToConsume)
                                 break;
                         }
                     } while (readNextMessage);
+
+                    if (options.Acknowledge == AcknowledgeBehavior.OnSuccess && receivedRawMessages.Count > 0)
+                    {
+                        foreach (var msg in receivedRawMessages)
+                            msg.Acknowledge();

```

Comment:
According to docs we can just ACK the last received message, not go through all of those. 
https://activemq.apache.org/components/nms/msdoc/1.6.0/vs2005/Output/html/T_Apache_NMS_AcknowledgementMode.htm

> With this acknowledgment mode, the client acknowledges a consumed message by calling the message's acknowledge method. This acknowledgement acknowledges the given message and all unacknowedged messages that have preceeded it for the session in which the message was delivered.